### PR TITLE
fix: added inline edit task

### PR DESCRIPTION
**Pull request by WINt1703**
fixes: #500

### CODE REVIEW

**Review by PascalHelbig — COMMENTED** [review]
Looks good to me, good job!

I have only two small comments left.

File: apps/web/frontend/components/taskList/taskCell.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  })
+
+  const handleSubmitTask = async (taskData: TaskFormData) => {
+    const result = await taskUpdate({
+      id: task.id,
+      data: {
+        projectId: task.project.id,
+        title: taskData.title,
+      },
+    })
```
Comment (PascalHelbig) [inline apps/web/frontend/components/taskList/taskCell.tsx:40]: I think it's good that the update got in here.

@WINt1703, what do you think about putting the deletion in here too?

File: apps/web/frontend/components/taskList/taskCell.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import { CgSpinner } from 'react-icons/cg'
+
+import { Button, InputField } from '@progwise/timebook-ui'
+
+import { TaskFragment, useTaskUpdateMutation } from '../../generated/graphql'
+import { TaskFormData, taskInputSchema } from './taskList'
+
+interface TaskCellProps {
+  task: TaskFragment
+  canDelete?: boolean
```
[the remark this answers is not in the document]
Comment (WINt1703) [reply apps/web/frontend/components/taskList/taskCell.tsx:14]: image
```TaskFragment```  doesn't have ```canDelete```

image

File: apps/web/frontend/components/taskList/taskCell.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+          />
+          {fetching && <CgSpinner className="inline h-8 w-8 animate-spin dark:text-blue-600" />}
+        </span>
+        <br />
+
+        <ErrorMessage errors={errors} name="title" as={<span role="alert" className="text-red-700" />} />
+      </div>
+
+      {openDeleteModal && (
+        // eslint-disable-next-line unicorn/no-useless-undefined
```
Comment (PascalHelbig) [inline apps/web/frontend/components/taskList/taskCell.tsx:71]: Which `undefined`

File: apps/web/frontend/components/taskList/taskCell.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (PascalHelbig) [reply apps/web/frontend/components/taskList/taskCell.tsx:16]: I am still not convinced. The `TaskList` gets tasks with `canModify`: https://github.com/progwise/timebook/blob/main/apps/web/frontend/components/taskList/taskList.tsx#L25

I think we can do here the same.

**Review by PascalHelbig — APPROVED** [review]
Nice!

**Pull request merged**